src/index.js

```diff
@@ -1,63 +1,123 @@
+        bookList.innerHTML += `
+    <li class="list-item" draggable="true">
+      <input type="checkbox" class="position-grid-start" id="${tasks.index}" checked/>
+      <p contenteditable="true" class="editable"> ${tasks.description} </p>
       <i class="fas fa-ellipsis-v position-grid-end"></i>
+      <i id="${tasks.index}" class="fas fa-trash-alt position-grid-end delete-button"></i>
     <li>   
     `;
-    } else {
-      bookList.innerHTML += `
-    <li class="list-item">
+      } else {
+        bookList.innerHTML += `
+    <li class="list-item" draggable="true">
     <input type="checkbox" class="position-grid-start" id="${tasks.index}"/>
-      <p> ${tasks.description} </p>
+      <p contenteditable="true" class="editable"> ${tasks.description} </p>
       <i class="fas fa-ellipsis-v position-grid-end"></i>
+      <i id="${tasks.index}" class="fas fa-trash-alt position-grid-end delete-button"></i>
     <li>   
     `;
+      }
+    });
+  }
+  bookList.innerHTML += '<button id="clear-button" class="clear-button"> Clear All Completed </button>';
+  document.querySelector('.delete-button').addEventListener('click', (event) => {
+    const { target } = event;
+    if (target.classList.value.indexOf('delete') !== -1) {
+      const tasksArray = JSON.parse(localStorage.getItem('tasksList'));
+      // eslint-disable-next-line max-len
+      const index = tasksArray.findIndex((element) => element.index === parseInt(target.id, 10));
+      // deleteTask(index, tasksArray);
+      // representNewIndexes(tasksArray);
+      tasksArray.splice(index, 1);
+      localStorage.setItem('tasksList', JSON.stringify(tasksArray));
+      // location.reload();
     }
   });
-  bookList.innerHTML += '<button class="clear-button"> Clear All Completed </button>';
 };
 
 document.addEventListener('DOMContentLoaded', () => {
   getLocalStorageTasks();
-  checkBoxEventListener(bookList, tasks);
+  checkBoxEventListener(bookList);
+  editTaskEventListener(bookList);
+  clearAllCompletedTasks(bookList);
 });
 document.addEventListener('DOMContentLoaded', populatelist);
+input.addEventListener('keypress', (e) => {
+  if (e.key === 'Enter') {
+    let newTask;
+    if (tasks) {
+      newTask = {
+        description: input.value,
+        completed: false,
+        index: tasks.length + 1,
+      };
+    } else {
+      newTask = {
+        description: input.value,
+        completed: false,
+        index: 0,
+      };
+    }
+    tasks = addNewTask(newTask, tasks);
+    pushLocalStorageTasks(tasks);
+    populatelist();
+    // eslint-disable-next-line no-restricted-globals
+    location.reload();
+  }
+});
+
+bookList.addEventListener('click', (event) => {
+  const { target } = event;
+  if (target.classList.value.indexOf('delete') !== -1) {
+    const tasksArray = JSON.parse(localStorage.getItem('tasksList'));
+    // eslint-disable-next-line max-len
+    const index = tasksArray.findIndex((element) => element.index === parseInt(target.id, 10));
```

Comment:
- [ ] Let us fix eslint errors without disabling them, this makes your code looks unprofessional.